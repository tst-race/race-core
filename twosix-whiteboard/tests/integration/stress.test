#!/bin/bash

DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
source "${DIR}"/helpers.sh

start_containers

# 10 megabyte string
LONG_STRING=$(cat /dev/urandom | tr -dc 'a-zA-Z0-9' | head -c 10M)

# TODO health check?
sleep 2

echo "post huge strings" | tee >(cat >&2)
(
for (( i = 0; i < 100; i++ )); do
    (
    curl --fail --silent --show-error --request POST localhost:5000/post/long --header 'Content-Type: application/json' -d @- <<CURL_DATA | jq -c  'del(.timestamp)'
{ "data": "$LONG_STRING" }
CURL_DATA
    ) &
done
) | sort

wait

echo "get index huge string (shouldn't exist becaue resizing)" | tee >(cat >&2)
curl --write-out "\n" --silent --show-error --request GET "localhost:5000/get/long/0" | wc

echo "get index huge string" | tee >(cat >&2)
curl --write-out "\n" --silent --show-error --request GET "localhost:5000/get/long/-1" | wc

echo "get range huge strings" | tee >(cat >&2)
curl --write-out "\n" --silent --show-error --request GET "localhost:5000/get/long/0/-1" | wc

stop_containers