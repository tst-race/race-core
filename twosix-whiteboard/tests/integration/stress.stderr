building docker image in case it's not up-to-date
create docker network if in case it doesn't exist
start containers
post huge strings
get index huge string (shouldn't exist becaue resizing)
get index huge string
get range huge strings
stop containers
